### PR TITLE
ref(api): split api models into a class per file

### PR DESCRIPTION
**Pull request by helgi**
Allows a separate of a class per file and makes it easier to read through the code, reason with what logic belongs to which functionality and generally work with it

See:
* https://code.djangoproject.com/wiki/CookBookSplitModelsToFiles
* http://paltman.com/breaking-apart-models-in-django/

### CODE REVIEW

**Comment by krancour** [comment]
+1 on this and the rationale for it.

**Comment by mboersma** [comment]
This is failing several unit tests: https://travis-ci.org/deis/workflow/builds/95955481

**Comment by helgi** [comment]
Yeah, it's the `_etcd_client` mocking and how imports are now interacting with one another. Trying to figure out a better way to do the mocking and see if there is a more sensible way to do `_etcd_client`

File: rootfs/api/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    return get_etcd_client.client
+
+
+# define update/delete callbacks for synchronizing
+# models with the configuration management backend
+
+def _etcd_purge_user(**kwargs):
+    try:
+        url = '/deis/builder/users/{}'.format(kwargs['instance'].username)
```
Comment (arschles) [inline rootfs/api/models/__init__.py]: not sure what the old code did, but does this need a check to see if `'instance'` exists in the dict?

File: rootfs/api/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# Load in all models
+from app import App  # noqa
+from build import Build  # noqa
+from certificate import Certificate  # noqa
+from config import Config  # noqa
+from container import Container  # noqa
+from domain import Domain  # noqa
+from key import Key  # noqa
+from push import Push  # noqa
+from release import Release  # noqa
```
Comment (arschles) [inline rootfs/api/models/__init__.py]: why not put these at the top? something to do with flake?

File: rootfs/api/models/__init__.py (hunk 1 above)
Comment (mboersma) [inline rootfs/api/models/__init__.py]: That would be good defensive coding practice, but in this case I think [Django signals](https://docs.djangoproject.com/en/1.6/ref/signals/#post-delete) effectively guarantee that `instance` will be there.

File: rootfs/api/models/__init__.py (hunk 2 above)
Comment (helgi) [inline rootfs/api/models/__init__.py]: Had to put them at the bottom or python would get angry me importing `get_etcd_client` into individual models. I could have left out those imports completely but for consistency (and smaller chance of other code breaking) I left imports in there with `# noqa` for flake

File: rootfs/api/models/__init__.py (hunk 1 above)
Comment (arschles) [inline rootfs/api/models/__init__.py]: :+1: cool

File: rootfs/api/models/__init__.py (hunk 2 above)
Comment (arschles) [inline rootfs/api/models/__init__.py]: got it

File: rootfs/api/models/app.py [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+        config = Config.objects.create(owner=self.owner, app=self)
+        Release.objects.create(version=1, owner=self.owner, app=self, config=config, build=None)
+
+    def delete(self, *args, **kwargs):
+        """Delete this application including all containers"""
+        try:
+            # attempt to remove containers from the scheduler
+            self._destroy_containers([c for c in self.container_set.exclude(type='run')])
+        except RuntimeError:
+            pass
```
Comment (arschles) [inline rootfs/api/models/app.py]: at least log here?

again, don't know what the old code did so not a blocker if this is copied

File: rootfs/api/models/app.py [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+
+            raise EnvironmentError('Error accessing deis-logger')
+
+        return r.content
+
+    def run(self, user, command):
+        """Run a one-off command in an ephemeral app container."""
+        if self.release_set.latest().build is None:
+            raise EnvironmentError('No build associated with this release to run this command')
+        # TODO: add support for interactive shell
```
Comment (arschles) [inline rootfs/api/models/app.py]: ticket for this? can't tell if this is a requirement or a nice-to-have

Comment (helgi) [inline rootfs/api/models/app.py]: This is from the old code - no action required for this PR

File: rootfs/api/models/app.py (hunk 3 above)
Comment (helgi) [inline rootfs/api/models/app.py]: Old code

File: rootfs/api/models/app.py (hunk 4 above)
Comment (arschles) [inline rootfs/api/models/app.py]: cool

**Comment by arschles** [comment]
@helgi :+1: on splitting up that huge file

**Pull request merged**